integrations/unstructured/src/haystack_integrations/components/converters/unstructured/converter.py

```diff
@@ -136,13 +148,16 @@ def _create_documents(
 
         if document_creation_mode == "one-doc-per-file":
             text = separator.join([str(el) for el in elements])
-            docs = [Document(content=text, meta={"name": str(filepath)})]
+            metadata = meta.copy()
+            metadata["name"] = str(filepath)
+            docs = [Document(content=text, meta=metadata)]
 
         elif document_creation_mode == "one-doc-per-page":
             texts_per_page: defaultdict[int, str] = defaultdict(str)
             meta_per_page: defaultdict[int, dict] = defaultdict(dict)
             for el in elements:
-                metadata = {"name": str(filepath)}
+                metadata = meta.copy()
```

Comment:
```suggestion
                metadata = copy.deepcopy(meta)
```
[To avoid problems with nested dictionaries](https://stackoverflow.com/a/2465951/10883094), let's make a deep copy.

Reply:
Did not know about that, good catch !